fastlane_core/lib/fastlane_core.rb

```diff
@@ -3,39 +3,40 @@
 require_relative 'fastlane_core/core_ext/string'
 require_relative 'fastlane_core/core_ext/shellwords'
 
+require_relative 'fastlane_core/analytics/action_completion_context'
+require_relative 'fastlane_core/analytics/action_launch_context'
+require_relative 'fastlane_core/analytics/analytics_event_builder'
+require_relative 'fastlane_core/analytics/analytics_ingester_client'
+require_relative 'fastlane_core/analytics/analytics_session'
+require_relative 'fastlane_core/build_watcher'
+require_relative 'fastlane_core/cert_checker'
+require_relative 'fastlane_core/clipboard'
+require_relative 'fastlane_core/command_executor'
+require_relative 'fastlane_core/configuration/configuration'
+require_relative 'fastlane_core/device_manager'
 require_relative 'fastlane_core/env'
+require_relative 'fastlane_core/fastlane_folder'
+require_relative 'fastlane_core/fastlane_pty'
 require_relative 'fastlane_core/feature/feature'
 require_relative 'fastlane_core/features'
 require_relative 'fastlane_core/helper'
-require_relative 'fastlane_core/configuration/configuration'
-require_relative 'fastlane_core/update_checker/update_checker'
-require_relative 'fastlane_core/languages'
-require_relative 'fastlane_core/cert_checker'
 require_relative 'fastlane_core/ipa_file_analyser'
+require_relative 'fastlane_core/ipa_upload_package_builder'
 require_relative 'fastlane_core/itunes_transporter'
-require_relative 'fastlane_core/provisioning_profile'
+require_relative 'fastlane_core/keychain_importer'
+require_relative 'fastlane_core/languages'
 require_relative 'fastlane_core/pkg_file_analyser'
 require_relative 'fastlane_core/pkg_upload_package_builder'
-require_relative 'fastlane_core/command_executor'
-require_relative 'fastlane_core/ipa_upload_package_builder'
 require_relative 'fastlane_core/print_table'
 require_relative 'fastlane_core/project'
-require_relative 'fastlane_core/device_manager'
-require_relative 'fastlane_core/ui/ui'
-require_relative 'fastlane_core/fastlane_folder'
-require_relative 'fastlane_core/keychain_importer'
+require_relative 'fastlane_core/provisioning_profile'
+require_relative 'fastlane_core/queue_worker'
 require_relative 'fastlane_core/swag'
-require_relative 'fastlane_core/build_watcher'
-require_relative 'fastlane_core/ui/errors'
-require_relative 'fastlane_core/test_parser'
-require_relative 'fastlane_core/analytics/action_completion_context'
-require_relative 'fastlane_core/analytics/action_launch_context'
-require_relative 'fastlane_core/analytics/analytics_event_builder'
-require_relative 'fastlane_core/analytics/analytics_ingester_client'
-require_relative 'fastlane_core/analytics/analytics_session'
 require_relative 'fastlane_core/tag_version'
-require_relative 'fastlane_core/fastlane_pty'
-require_relative 'fastlane_core/queue_worker'
+require_relative 'fastlane_core/test_parser'
+require_relative 'fastlane_core/ui/errors'
+require_relative 'fastlane_core/ui/ui'
+require_relative 'fastlane_core/update_checker/update_checker'
```

Comment:
I like reorganising dependencies alphabetically, but it may have some unintended side-effects in Ruby.
This PR works on my machine, but let's wait for CircleCI's unit tests to complete before merging.

Reply:
As long as tests pass, is this safe? I have no clue 😮 what are/were the risks? maybe cc @ainame 🙏 😁

Reply:
CircleCI passed, so it should be safe ✅

As you saw in [fastlane_core.rb](https://github.com/fastlane/fastlane/blob/b2b214f4a6d2450bdefff4dd36d50df79471b51a/fastlane_core/lib/fastlane_core.rb), some requires have to be ordered before:
```ruby
# Ruby monkey-patches - should be before almost all else
require_relative 'fastlane_core/core_ext/string'
require_relative 'fastlane_core/core_ext/shellwords'
```
And some after:
```ruby
require 'commander'

# after commander import
require_relative 'fastlane_core/ui/fastlane_runner' # monkey patch
```

They are usually extensions of Ruby classes, without those we'll probably get `NoMethodError`s in CircleCI.

Reply:
As long as test cases cover those files, it would be fine🙂 Even in the monkey patch scenario mentioned, it doesn't matter as long as Ruby doesn't evaluate such patched methods directly. 

For example,  let's say we need to load two files in `fastlane_core.rb` and `file A` is like below, it is safe. The order doesn't matter. `monkey_patch_method_return_true` will be evaluated after everything loaded. 

```ruby
# fastlane_core.rb
require 'file A'
require 'monkey_patch'
```

```ruby
# file A' 
class A 
  def hello
    puts("hello") if "some_string".monkey_patch_method_return_true
  end
end
```

However, `file A'` causes an error because we need to `require` file 'moneky_patch' that implements `monkey_patch_method_return_true` to String before `file A` is loaded. 

```ruby
# file A
if "some_string".monkey_patch_method_return_true
  class A 
    def hello; end
  end
end
```

`commander` scenario surely depends on the gem to define the classes first and then monkey patching follows.

You can test whether that file loading works by `ruby fasltane_core/lib/fastlane_core.rb`. Even so, there is a small chance order updates create logic-level discrepancy but unit tests should cover that 🤞 

---

It's off-topic but `fastlane`'s unit testing runs test cases exactly the same order every time right now so combining existing `require`  order and unit test cases `order` may create a happy code path that still leaves some edge-cases technically😬  

I've been trying to improve that in a PR... 
https://github.com/fastlane/fastlane/pull/18278

Reply:
This slightly terrifies me but I do prefer it this way 😬 I don’t _think_ this should causes any issues but still a bit spooky 🤷‍♂️